docs/ops/doc/CallingOps.md

```diff
@@ -1,61 +1,96 @@
+With the name established in the `OpBuilder` chain, we can then specify our input(s) with the `.input()` method.
 
-For our gaussian blur, we will pass as inputs our input image `inImage`, and a `double` as our sigma parameter:
+For this Gaussian blur, we have two inputs: `inImage` is the image we want to blur, and a `double` value as our sigma parameter:
 
 ```groovy
 ops.op("filter.gauss").input(inImage, 2.0)
 ```
 
 ## Passing an output buffer with `.output()`
 
-Now that the inputs are specified, we can chain the output buffer using the `output()` method.
+After specifying inputs, we provide a preallocated output container using the `.output()` method.
 
-For our gaussian blur, we will pass as the output buffer our output image `outImage`:
+For our Gaussian blur, we will pass our output image `outImage` as a receptacle for the result:
 
 ```groovy
 ops.op("filter.gauss").input(inImage, 2.0).output(outImage)
 ```
 
 ## Computing with `.compute()`
 
-With all of the components of the needed Op specified, we can begin computation with the `.compute()` method.
+With all of our desired Op's inputs and output now specified, we can run it with the `.compute()` method.
 
 ```groovy
 ops.op("filter.gauss").input(inImage, 2.0).output(outImage).compute()
 ```
 
-In the call to `compute()`, the `OpEnvironment` will use the components of the `OpBuilder` syntax to:
-* Match an Op based on the name provided, as well as the types of the provided input and output `Object`s
-* Execute the Op on the provided input and output `Object`s.
+In the call to `compute()`, the `OpEnvironment` will use all of the parameters provided to:
+* Match an Op based on the name provided, as well as the types of the provided input and output objects
+* Execute the Op on the provided input and output objects.
+
+After this step, `outImage` will contain the results of the Gaussian blur on `inImage`.
+
+## Variations on use
 
-## Additions: Repeating execution
+### Using a *function* or *inplace* Op
 
-When an Op should be executed many times on different inputs, the `OpBuilder` syntax can be modified to return the *Op* instead. Instead of calling the `.compute()` function at the end of our `OpBuilder` call, we can instead call the `.computer()` method to get back the matched Op:
+Calling our Gaussian blur as a *computer* above is great when we have pre-allocated output, but for other scenarios we can request Ops as *functions* or *inplaces*.
+
+*Functions* are used when we want to *create* the final output, indicated by ending the builder with `.apply()`:
+
+```groovy
+var outImage = ops.op("filter.gauss").input(inImage, 2.0).apply()
+```
+
+*Inplaces* are used when we want to destructively modify one of the existing inputs (which is explicitly forbidden by *computers*; a *computer* Op's output should be a different object from all of its inputs). We indicate this by the `mutate#()` method, where the `#` corresponds to the *parameter index* that will be modified:
```

Comment:
Would it be confusing, or clarifying, to note this is 1-indexed?

Reply:
06c20936aeca474ffed93ed2cdbf771bdeaa2f05